### PR TITLE
Issue/1813 support part3

### PR DESCRIPTION
**Pull request by valllllll2000**
#### The solution
final solution to send full device information to support.
How does it work?
In a support ticket we realize that we need device information from one or more devices. We inform the partner to request that info directly from the device. In order to send it they will need the ref id, see screenshot
Screenshot from 2021-07-30 10-31-27

The device user will go to settings and find "send device info to support" and press that option. This will open a dialog and he then can write the 4 digits reference id. 
#### Reviewer Checklist

* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header
* [ ] Formatted the code per our style guide
* [ ] Added a documentation (if relevant)

### CODE REVIEW

File: app/src/main/java/org/akvo/flow/presentation/settings/SendDeviceInfoDialog.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2018-2019 Stichting Akvo (Akvo Foundation)
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/presentation/settings/SendDeviceInfoDialog.kt:2]: Is it a new file?

File: support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt [hunk 2]
```diff
@@ -57,29 +55,27 @@ class SendConversation @Inject constructor(
...
+            while (page < pageCount) {
                 conversationsResult = restApi.fetchConversations(page)
                 conversations = conversationsResult.conversations
-                totalCount += conversationsResult.totalCount
-                countLoaded = conversations?.size ?: 0
                 conversation = findConversation(conversations, refId)
                 if (conversation != null) {
                     return conversation
                 }
+                page++
```
Comment (muloem) [inline support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt:78]: Here maybe you want to do something like a `do  { ... } while ()`?

Comment (muloem) [reply support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt:78]: to make it simpler to read..

File: app/src/main/java/org/akvo/flow/presentation/settings/SendDeviceInfoDialog.kt (hunk 1 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/presentation/settings/SendDeviceInfoDialog.kt]: no, I had copied from another file and forgot to update the copyright

File: support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt (hunk 2 above)
Comment (valllllll2000) [reply support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt]: done!

File: support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt [hunk 3]
```diff
@@ -57,30 +55,27 @@ class SendConversation @Inject constructor(
...
         var conversations = conversationsResult.conversations
-        var totalCount = conversationsResult.totalCount
-        var countLoaded = conversations?.size ?: 0
+        val pageCount = conversationsResult.pageCount
 
-        val refId = parameters[PARAM_REF_ID] as Int
         var conversation = findConversation(conversations, refId)
         if (conversation != null) {
             return conversation
         } else {
```
Comment (muloem) [inline support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt:68]: Isnt it possible to have all this within the `do ... while()`?

Comment (valllllll2000) [reply support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt]: done!

**Review by muloem — APPROVED** [review]

**Pull request merged**